app/Http/Controllers/SessionController.php

```diff
@@ -63,24 +62,16 @@ public function init(): array
 				// If the session is unauthenticated ('user' === null), but grants admin rights nonetheless,
 				// the front-end shows the dialog to create an admin account.
 				$return['user'] = null;
-				$return['rights'] = [
-					'is_admin' => true,
-					'is_locked' => false,
-					'may_upload' => true,
-				];
+				$return['rights'] = new UserRights(true, true, true);
```

Comment:
This is the issue which still bothers me most, because the attributes here are still not rights, but capabilities.

I know that I named this objects `'rights'` and that it was me who put the user capabilities into it mostly because I did not want to change too much here and in the frontend at the time when I cleaned up the `init()` method. But now, when were are cleaning up the naming convention for rights, capabilities etc. what this PR intends to we should get it right.

Rights are something based on the relation between the currently authenticated user and another object. As I already said above rights of a user on a user would be things like `MAY_CHANGE_PASSWORD`, i.e. may user X change the password of user Y.

When I added this line I envisioned a two-level array like this:

```php
$return['rights'] = [
    'root_album' => [
        can_edit => true,
        can_upload => true,
        can_download => true,
        /* more rights on root album */
    ],
    'settings' => [
        'can_edit' => true,
        /* more detailed rights on settings in the future */
    ],
   /* more objects on which a user can have rights */
];
```

This approach would stress the relational nature of rights and clearly, things like `may_administrate` would not be among them.

Given the new development above, I wonder whether we still need this entry at all or if we are just fine the new DTO above. So lets step through the three attributes we currently have in `UserRights`:

 - `can_upload_root`: I assume we don't need this here at all. The same way as albums and photos are wrapped into a DTO `AlbumDTO` and `PhotoDTO` we could as well wrap the root album into one which augments the rights of the user on the root album.
 - `can_edit_own_settings`: Not needed for the same reason. Instead of returning the user model one line above we should return a DTO which wraps the user model and augments rights of the current user on that user (which happens to be the user itself). At the moment I see to actual user rights (on users): `can_edit_all` and `can_edit_password`.
 - `can_administrate`: This one is a little problematic right now without changing too much in the frontend as well. But basically it could and should be solved in the same spirit. For the time being (i.e. without changing too much in the frontend) we may as well drop this one (for now) and directly use the capability `may_administrate` of the user model in the frontend. (This is also not the perfect solution, but still better than calling this one rights here.)

------------------------------

However, there is one more problem to solve and this is how we express rights on `null` models. Currently, we need this in two cases: for the admin user as long as no real admin user exists (i.e. after installation) and for the root album. Currently, the DTOs above (i.e. `AlbumDTO` and `PhotoDTO` take a non-null model and _add_ the attribute `rights` to it. This approach does not work if the model itself is `null`. Hence, maybe a more sustainable approach would be not to _add_ the attribute, but place it next to the model, i.e. something like that

```php
class ModelWithRighsDTO extends ArrayableDTO {
    public ?Model $model;
    public ModelRights $rights;

    public function __construct(?Model $model = null) {
        $this->model = $model;
        $this->rights = new ModelRights::ofModel($model);
    }
}
```

This would also allow to report rights on `null` models to the frontend and solve a lot problems.